src/App.jsx

```diff
@@ -1,23 +1,44 @@
-import React from 'react';
-import { Routes, Route, Navigate } from 'react-router-dom';
+import React, { useEffect, useState } from 'react';
+import { Routes, Route, Navigate, useNavigate, useLocation } from 'react-router-dom';
+import { store } from './redux/store';
 
 import Notes from './pages/myNotes/notes';
 import Header from './pages/header/header';
 import SharedNotes from './pages/sharedNotes/SharedNotes';
-
+import SignUp from './pages/signIn/signUp';
 import About from './pages/about/About';
 import NotFound from './pages/notFound/notFound';
+import SignIn from './pages/signIn/SignIn';
 
 function App() {
+    const location = useLocation();
+    const navigate = useNavigate();
+    const [hasAccess, setHasAccess] = useState(false);
+    store.subscribe(() => setHasAccess(store.getState().userInfo.access));
+    useEffect(() => {
+        setHasAccess(store.getState().userInfo.access);
```

Comment:
you call the part "store.getState().userInfo.access" 2 times. You can call it once and use in both cases